apps/src/lib/client/rpc.rs

```diff
@@ -1104,45 +1319,77 @@ pub async fn query_shielded_balance(
                     .shielded
                     .decode_amount(client.clone(), balance, epoch)
                     .await;
-                print_decoded_balance(ctx, decoded_balance);
+                print_decoded_balance(ctx, &client, decoded_balance).await;
             }
         }
     }
 }
 
-pub fn print_decoded_balance(
+pub async fn print_decoded_balance(
     ctx: &mut Context,
-    decoded_balance: Amount<Address>,
+    client: &HttpClient,
+    decoded_balance: MaspDenominatedAmount,
```

Comment:
This approach is valid. However this function could instead take an `Amount<Address, i256>`. That would relieve this function from having to aggregate `u64` denominations into a `u256`. This should be straightforward to do if functions like `decode_amount` also returned `Amount<Address, i256>`s.